lib/statsd/instrument.rb

```diff
@@ -130,6 +130,12 @@ def statsd_count_success(method, name, sample_rate: nil, tags: nil, no_prefix: f
           client ||= StatsD.singleton_client
           suffix = truthiness == false ? "failure" : "success"
           key = StatsD::Instrument.generate_metric_name(name, self, *args)
+
+          if tag_error_class && error
+            tags ||= {}
+            tags[:error_class] = error.class.name
```

Comment:
Actually there's a small issue here. `tags:` is passed once, so we shouldn't mutate it.

You need to `dup` it first.